### PR TITLE
OCPBUGS-1131: [release-4.11] e2e: perfprof: unbreak the e2e-gcp PAO lane

### PR DESCRIPTION
**Pull request by marioferh**
Manual cherry-pick of https://github.com/openshift/cluster-node-tuning-operator/pull/438

e2e: add podsecurity labels, disable OCP label sync

Backport and squash two commits related to handling
of security labels:

b0024584fb244691097916665d5cec1e7c1dc5f4 (partial):    
    OCP >= 4.12 wants to have stricter podsecurity rules.
    In our e2e tests we do a bunch of stuff, including running privileged
    pods. We just annotate the test namespace(s) to signal we need top
    privileges. Since e2e tests run in very controlled envs (CI mostly),
    and since test namespace should be gone anyway once the e2e tests
    are finished, this is still fair game.
    
    Signed-off-by: Francesco Romani <fromani@redhat.com>

b0024584fb244691097916665d5cec1e7c1dc5f4
    podsecurity: disable OCP label sync

    Per last OCP recommendation (internal doc) is not sufficient to
    declare the pod security labels, we also need to disable the
    label sync, which we do here.

    Signed-off-by: Francesco Romani <fromani@redhat.com>
    
Co-Authored-by: Mario Fernandez <mariofer@redhat.com>
Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

File: test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: that's not the original commit, would have been (much much) better to note that in the commit message, mentioning the other one squashed here

**Review by fromanirh — COMMENTED** [review]
/approve

File: test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go
Comment (fromanirh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: something like:
```
e2e: add podsecurity labels, disable OCP label sync

Backport and squash two commits related to handling
of security labels:

b0024584fb244691097916665d5cec1e7c1dc5f4 (partial):    
    OCP >= 4.12 wants to have stricter podsecurity rules.
    In our e2e tests we do a bunch of stuff, including running privileged
    pods. We just annotate the test namespace(s) to signal we need top
    privileges. Since e2e tests run in very controlled envs (CI mostly),
    and since test namespace should be gone anyway once the e2e tests
    are finished, this is still fair game.
    
    Signed-off-by: Francesco Romani <fromani@redhat.com>

b0024584fb244691097916665d5cec1e7c1dc5f4
    podsecurity: disable OCP label sync

    Per last OCP recommendation (internal doc) is not sufficient to
    declare the pod security labels, we also need to disable the
    label sync, which we do here.

    Signed-off-by: Francesco Romani <fromani@redhat.com>
    
Co-Authored-by: Mario Fernandez <mariofer@redhat.com>
Signed-off-by: Francesco Romani <fromani@redhat.com>
```
(please doublecheck the commit hashes)

Comment (fromanirh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: thanks for fixing the PR description. It would have been much better to fix the relevant commit message AND THEN the PR description, but I guess we can move forward this time. Let's evaluate deeper next time.

Comment (marioferh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: I can do it but whats is the purpose of that if we squash the commits?

Comment (fromanirh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: we should not squash the commits when doing backports - we should have as much as possible the same commits as the main branch, otherwise half of the purpose of doing backports is lost.
In general, we should preserve the authorship properly and the commit boundary as much as possible. There's almost never a good reason not to.
That becomes critical if contributions are from different organizations.

Comment (marioferh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: Get it.

Comment (MarSik) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: Folks, the tests passed, @fromanirh can you add lgtm if all is fine now?

Comment (fromanirh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: > Folks, the tests passed, @fromanirh can you add lgtm if all is fine now?

I was waiting for other reviews.

Comment (marioferh) [reply test/e2e/performanceprofile/functests/utils/namespaces/namespaces.go:33]: @yanirq @swatisehgal PTAL?

**Review by MarSik — APPROVED** [review]
/lgtm

**Pull request merged**
